packages/server/src/login/login-course.service.ts

```diff
@@ -31,38 +31,38 @@ export class LoginCourseService {
     const userCourses = [];
 
     for (const c of info.courses) {
-      const course: CourseModel = await this.courseSectionToCourse(
-        c.course,
-        c.section,
-      );
+      const course: CourseModel = await this.courseCRNToCourse(c.crn);
 
       if (course) {
         const userCourse = await this.courseToUserCourse(
           user.id,
           course.id,
-          Role.STUDENT,
+          c.role,
         );
         userCourses.push(userCourse);
       }
     }
 
-    if (info.ta_courses) {
-      for (const c of info.ta_courses) {
-        // Query for all the courses which match the name of the generic course from Khoury
-        const courseMappings = (
-          await CourseSectionMappingModel.find({
-            where: { genericCourseName: c.course }, // TODO: Add semester support
-            relations: ['course'],
-          })
-        ).filter((cm) => cm.course.enabled);
+    if (info.prof_courses) {
+      // loop through section groups
+      for (const c of info.prof_courses) {
+        // loop through CRNs to create courses
+        for (const courseCRN of c.crns) {
+          const courseMappings = (
+            await CourseSectionMappingModel.find({
+              where: { crn: courseCRN }, // TODO: Add semester support
+              relations: ['course'],
+            })
+          ).filter((cm) => cm.course.enabled);
```

Comment:
It looks like the logic here is the same as what's happening in `courseCrnToCourse`, so you can probably just call that function. Also I believe you only need to inspect one of the CRNs here - all of them should map to the same CourseModel since they're all for the same section group.